src/test/osd/RadosModel.h

```diff
@@ -1948,6 +1958,224 @@ class CopyFromOp : public TestOp {
   }
 };
 
+class SetRedirectOp : public TestOp {
+public:
+  string oid, oid_tgt, tgt_pool_name;
+  ObjectDesc src_value, tgt_value;
+  librados::ObjectWriteOperation op;
+  librados::ObjectReadOperation rd_op;
+  librados::AioCompletion *comp;
+  ceph::shared_ptr<int> in_use;
+  int done;
+  int r;
+  SetRedirectOp(int n,
+	     RadosTestContext *context,
+	     const string &oid,
+	     const string &oid_tgt,
+	     const string &tgt_pool_name,
+	     TestOpStat *stat = 0)
+    : TestOp(n, context, stat),
+      oid(oid), oid_tgt(oid_tgt), tgt_pool_name(tgt_pool_name),
+      comp(NULL), done(0), 
+      r(0)
+  {}
+
+  void _begin() override
+  {
+    Mutex::Locker l(context->state_lock);
+    context->oid_in_use.insert(oid);
+    context->oid_not_in_use.erase(oid);
+    context->oid_redirect_in_use.insert(oid_tgt);
+    context->oid_redirect_not_in_use.erase(oid_tgt);
+
+    context->find_object(oid, &src_value); 
+    if(!context->redirect_objs[oid].empty()) {
+      /* update target's user_version */
+      rd_op.stat(NULL, NULL, NULL);
+      comp = context->rados.aio_create_completion();
+      context->io_ctx.aio_operate(context->prefix+oid_tgt, comp, &rd_op,
+			    librados::OPERATION_ORDER_READS_WRITES,
+			    NULL);
+      comp->wait_for_safe();
+      context->update_object_version(oid_tgt, comp->get_version64());
```

Comment:
We use two object pools (A. objects for normal OP  B. objects for redirect target ).
Therefore, A indicate B.
In this case, copy_from will only update A's version (not B).